gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -811,6 +811,17 @@ subroutine Initialize ( gc, import, dumexport, clock, rc )
                                       label=trim(string) // 'ref_time:',_RC )
        _ASSERT(is_valid_time(list(n)%ref_time),'Invalid ref_time')
 
+       call ESMF_ConfigGetAttribute ( cfg, list(n)%start_date, default=-999, &
+                                      label=trim(string) // 'start_date:',_RC )
+       if (list(n)%start_date /= -999) then
+          _ASSERT(is_valid_date(list(n)%start_date),'Invalid start_date')
+       end if
```

Comment:
Can checking for `-999` (magic value!) be moved inside the `is_valid_date()` procedure?  If so, the logic will be much cleaner.

And ... if you are able to make that change, then please introduce a named parameter up top for "-999" and propagate in this file.

Reply:
Just adopted your suggestions